### PR TITLE
Remove 1000 char xattr limit

### PR DESCRIPTION
**Pull request by crrodriguez**
We need to access xattrs much bigger than 1000, up to the size
allowed by the filesystem.

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
@crrodriguez please have a look at the failing tests - to my understanding the tests should still all be green.
In addition adding a test for xattr wiht more then 1000 chars would be a good thing. THX

**Comment by eduardok** [comment]
Please try this code instead: https://github.com/eduardok/libsmbclient-php/tree/edtests

**Comment by crrodriguez** [comment]
> Please try this code instead: https://github.com/eduardok/libsmbclient-php/tree/edtests

This patch is the same thing I had before, but avoided posting it do the different paths needed for old PHP versions, should do exacmle the same thing.

**Comment by eduardok** [comment]
> > Please try this code instead: https://github.com/eduardok/libsmbclient-php/tree/edtests
> 
> This patch is the same thing I had before, but avoided posting it do the different paths needed for old PHP versions, should do exacmle the same thing.

Not sure what you mean by exactly the same since this pull request's code (https://github.com/eduardok/libsmbclient-php/pull/76/commits/b8137f6617b14ce69d4992a136063439f8bf4c7e) is quite different from the one I pointed you to (https://github.com/eduardok/libsmbclient-php/commit/565d0afd3201b7ba797ee195fb801fb6909a45bb). It uses no ALLOCA_FLAG/do_alloca, handles different PHP versions, and passes CI (https://travis-ci.org/github/eduardok/libsmbclient-php/builds/748486644).

**Comment by crrodriguez** [comment]
> > > Please try this code instead: https://github.com/eduardok/libsmbclient-php/tree/edtests
> > 
> > 
> > This patch is the same thing I had before, but avoided posting it do the different paths needed for old PHP versions, should do exacmle the same thing.
> 
> Not sure what you mean by exactly the same since this pull request's code ([b8137f6](https://github.com/eduardok/libsmbclient-php/commit/b8137f6617b14ce69d4992a136063439f8bf4c7e)) is quite different from the one I pointed you to ([565d0af](https://github.com/eduardok/libsmbclient-php/commit/565d0afd3201b7ba797ee195fb801fb6909a45bb)). It uses no ALLOCA_FLAG/do_alloca, handles different PHP versions, and passes CI (https://travis-ci.org/github/eduardok/libsmbclient-php/builds/748486644).

Sorry if I was unclear, I mean an early version of this request that isn't online was very similar to yours.

File: smbclient.c [hunk 1]
```diff
@@ -1649,20 +1651,28 @@ PHP_FUNCTION(smbclient_getxattr)
...
+	}
+
+	values = do_alloca(xattr_size + 1, use_heap);
+
+	retsize = smbc_getxattr(state->ctx, url, name, values, sizeof(values));
+
+	if(retsize < 0)
+		goto fail;
+
+	RETVAL_STRING(values);
```
Comment (remicollet) [inline smbclient.c:1672]: This will break support for PHP 5, see deleted line above

File: smbclient.c [hunk 2]
```diff
@@ -1649,20 +1651,28 @@ PHP_FUNCTION(smbclient_getxattr)
...
+
+	if(xattr_size < 0) {
+		goto fail;
 	}
+
+	if(xattr_size == 0) {
+		RETURN_EMPTY_STRING();
+	}
+
+	values = do_alloca(xattr_size + 1, use_heap);
```
Comment (remicollet) [inline smbclient.c:1665]: why not a simple emalloc ?

BTW, for PHP 7+, will be better to use a zend_string, which will avoid double allocation (RETURN_STR will use the allocated string, while RETURN_STRING allocate a new one)

File: smbclient.c [hunk 3]
```diff
@@ -1649,20 +1651,28 @@ PHP_FUNCTION(smbclient_getxattr)
...
+
+	retsize = smbc_getxattr(state->ctx, url, name, values, sizeof(values));
```
Comment (remicollet) [inline smbclient.c:1667]: sizeof(values) is only the pointer size...

**Comment by remicollet** [comment]
PR #96 seems better

**Comment by eduardok** [comment]
Pulled Remi's request, please use release 1.0.7.

**Comment by remicollet** [comment]
> We need to access xattrs much bigger than 1000, up to the size allowed by the filesystem.

Any idea of what size is needed ?
because dyn alloc doesn't work anymore :(

**Comment by crrodriguez** [comment]
> > We need to access xattrs much bigger than 1000, up to the size allowed by the filesystem.
> 
> Any idea of what size is needed ? because dyn alloc doesn't work anymore :(

64k

**Comment by remicollet** [comment]
@crrodriguez thanks (test or PR #100 welcome)

**Pull request closed without merging**